sw/device/tests/entropy_src_fw_override_test.c

```diff
@@ -360,13 +378,29 @@ status_t firmware_override_extract_insert(
 
   // Generate some random numbers.
   LOG_INFO("Generating random numbers...");
+  uint32_t rnd_timeout_usec = kRandomNumberTimeoutUsec;
+  if (kDeviceType == kDeviceSimVerilator) {
+    // Simulation on Verilator generates entropy much more slowly than other
+    // execution environments. As a result, the timeout to generate random
+    // numbers is increased only when running on Verilator.
+    rnd_timeout_usec = kVerilatorRandomNumberTimeoutUsec;
+  }
   uint32_t data;
-  for (size_t i = 0; i < 16; i++) {
+  for (size_t i = 0; i < kRandomNumberCount; i++) {
     TRY(rv_core_ibex_testutils_get_rnd_data(&rv_core_ibex,
-                                            /*timeout_usec=*/100 * 1000,
+                                            /*timeout_usec=*/rnd_timeout_usec,
                                             &data));
   }
 
+  if (kDeviceType == kDeviceSimVerilator) {
+    // If running on Verilator then entropy is generated much more slowly.
+    // It would take an impractical amount of time to generate the thousands
+    // of words of entropy that are required by all entropy consumers, and so
+    // since we are not concerned with the rate of entropy on Verilator, we
+    // stop at just generating some random numbers only.
+    return OK_STATUS();
+  }
```

Comment:
I think this is non-ideal because it means we skip quite a big portion of the main test function in Verilator. So the Verilator simulation becomes less useful for debugging as a big part of the design is not exercised anymore.

Another way to speed up the Verilator simulation would be to reduce the number of modes tested. E.g. for Verilator only we could change:
```c
  // Test all modes.
  static dif_entropy_src_single_bit_mode_t kModes[] = {
      kDifEntropySrcSingleBitModeDisabled, kDifEntropySrcSingleBitMode0,
      kDifEntropySrcSingleBitMode1,        kDifEntropySrcSingleBitMode2,
      kDifEntropySrcSingleBitMode3,
  };
```
to
```c
  // Test all modes.
  static dif_entropy_src_single_bit_mode_t kModes[] = {
      kDifEntropySrcSingleBitModeDisabled, kDifEntropySrcSingleBitMode0,
  };
```
This would just check 2 instead of 5 modes (..Mode0 / .. ..Mode3 are almost identical and all of them are 4x slower compared to ...ModeDisabled).

Also, the final phase:
```c
  // Rerun the test with single bit mode disabled,
  // this time with an output delay.
  fw_ov_insert_wait_enabled = true;
  EXECUTE_TEST(test_result, firmware_override_extract_insert,
               kDifEntropySrcSingleBitModeDisabled, false, false);
  EXECUTE_TEST(test_result, firmware_override_extract_insert,
               kDifEntropySrcSingleBitModeDisabled, true, false);
```
Could be skipped for Verilator. We don't learn a lot from this.

Reply:
Yes, I agree, this was one thing that I was not sure on.

Making your suggested changes (just testing the first two modes and skipping the final phase on Verilator), alongside the same timeout changes as before, causes the test to pass in 6656.2 seconds (around 1 hour 50 minutes) in an environment similar to CI. It is not clear to me whether properly tuning the AST model as suggested would change this.

This seems fine to me for debug purposes - it is far better than the 15+ hours it was taking before. Ideally we would like it to run in under an hour, but given the purpose of this test I think it makes more sense to tag the Verilator environment as manual for `entropy_src_fw_override_test` with a comment explaining the runtime, alongside these changes. Does this sound reasonable to you @vogelpi?

Reply:
Thanks for trying this out and reporting back. Tuning the AST model would help on top of this but we don't have to do this right now.

> This seems fine to me for debug purposes - it is far better than the 15+ hours it was taking before. Ideally we would like it to run in under an hour, but given the purpose of this test I think it makes more sense to tag the Verilator environment as manual for entropy_src_fw_override_test with a comment explaining the runtime, alongside these changes. Does this sound reasonable to you @vogelpi?

This would be perfect. Can you implement these changes directly in this PR?